### PR TITLE
GF-13, GF-17 - React Route Dom and Project page

### PR DESCRIPTION
**Pull request by iamdavidpetric**

### CODE REVIEW

File: src/show/Application.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
-import { Home } from './pages';
+import { Home, Project } from './pages';
 import { Navbar } from '../show/components';
+import { BrowserRouter, Routes, Route } from 'react-router-dom';
```
Comment (iamalextodoran) [inline src/show/Application.js]: Group by external libraries first then by in-house made stuff.

File: src/show/Application.js [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
-      <Home />
+      <div>
+        <Navbar />
+        <BrowserRouter>
+          <Routes>
+            <Route path='/' element={<Home />} />
+            <Route path='/project' element={<Project />} />
+          </Routes>
+        </BrowserRouter>
+      </div>
```
Comment (iamalextodoran) [inline src/show/Application.js]: Why do you need to wrap those into a div?

File: src/show/pages/public/Project/index.js [hunk 3]
```diff
@@ -0,0 +1,195 @@
+import { Button, ProgressBar } from '../../../components';
+import { FaFacebook, FaInstagram } from 'react-icons/fa';
+import { FiTwitter } from 'react-icons/fi';
+import { AiOutlineLink } from 'react-icons/ai';
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: GROUP AND SORT

File: src/show/pages/public/Project/index.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+            <img
+              src='https://i.imgur.com/kHgmAGw.jpg'
+              alt='bacpack'
+              className='object-cover rounded-lg h-full w-full mx-5 my-5'
+            />
+          </div>
+
+          <div className='mr-5 h-28 mt-5'>
+            <div className='flex ml-4 h-[112px] no-scrollbar overflow-x-scroll'>
+              {[...Array(40)].map(item => (
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Define a const (e.g. const projectPictures or const pictures) and use it.

File: src/show/pages/public/Project/index.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+import { AiOutlineLink } from 'react-icons/ai';
+
+const Project = () => {
+  return (
+    <div className='flex flex-col bg-white h-full w-full'>
+      <div className='flex bg-white h-20 mt-5'>
+        <div className='flex h-20 w-20 justify-center items-center'>
+          <div className='flex items-center justify-center w-16 h-16 rounded-full ml-5'>
+            <img
+              src='https://upload.wikimedia.org/wikipedia/commons/d/d8/Elon_Musk_Royal_Society_%28cropped%29.jpg'
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Define in a const

File: src/show/pages/public/Project/index.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+              alt='Elon Musk'
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Define in a const

File: src/show/pages/public/Project/index.js [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+              className='rounded-full object-cover h-14 w-14'
+            />
+          </div>
+        </div>
+        <div className='flex flex-col justify-center ml-2 w-full text-2xl'>
+          Hatz Musk "change your lifestyle with this product or another bs"
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Define in a const

File: src/show/pages/public/Project/index.js [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+        </div>
+      </div>
+
+      <div className='flex mt-5 mb-10'>
+        <div className='flex flex-col h-112 w-3/5'>
+          <div className='flex justify-center items-center h-full w-full'>
+            <img
+              src='https://i.imgur.com/kHgmAGw.jpg'
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Define in a const

File: src/show/pages/public/Project/index.js [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+              alt='bacpack'
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js:86]: Define in a const

File: src/show/pages/public/Project/index.js [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+                />
+              ))}
+            </div>
+          </div>
+        </div>
+
+        <div className='flex flex-col w-2/5 h-112 mx-10'>
+          <div>
+            <div className='flex flex-row h-12 text-2xl items-center justify-center '>
+              All Access | Outdoor Leisure Backpack
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Define in a const

File: src/show/pages/public/Project/index.js [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+            <div className='px-5 mt-2 text-gray-500'>
+              'All Access' aim is to remove the stresses of packing and
+              organising for outdoor activities so that people can souly enjoy
+              the positives of an active outdoor lifestyle. All the prototyping
+              and research of the product have already been carried out, so the
+              purpose of the funds are to finalise the design and begin
+              manufacture as soon as possible Once the minimum target has been
+              reached, a website will be made available to pre-order All Access
+              backpacks until a reliable manufacturer has been sourced, after
+              which shipping will begin right away.'
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Define in a const

File: src/show/pages/public/Project/index.js [hunk 12]
```diff
@@ -0,0 +1,195 @@
...
+            </div>
+            <div className='mt-5 px-5 text-lg text-center'>
+              25.000$/100.000$
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Define in a const

**Review by iamalextodoran — CHANGES_REQUESTED** [review]
Define consts just like you would fetch your data from the backend, use objects, arrays etc.

File: src/show/pages/public/Project/index.js [hunk 13]
```diff
@@ -0,0 +1,212 @@
+import { FiTwitter } from 'react-icons/fi';
+import { AiOutlineLink } from 'react-icons/ai';
+import { FaFacebook, FaInstagram } from 'react-icons/fa';
+
+import { Button, ProgressBar } from '../../../components';
+
+const project_details = {
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: camel case for this

File: src/show/pages/public/Project/index.js [hunk 14]
```diff
@@ -0,0 +1,212 @@
...
+  title: 'Backpack 7 in 1',
+  description:
+    'aim is to remove the stresses of packing and organising for outdoor activities so that people can souly enjoy the positives of an active outdoor lifestyle. All the prototyping and research of the product have already been carried out, so the purpose of the funds are to finalise the design and begin manufacture as soon as possible Once the minimum target has been reached, a website will be made available to pre-order All Access backpacks until a reliable manufacturer has been sourced, after which shipping will begin right away.',
+  statement: 'Hai jupana in hora',
+  team_members: ['Mihai', 'Petru', 'Alecs', 'Avram'],
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: this should be an array of objects.

File: src/show/pages/public/Project/index.js [hunk 15]
```diff
@@ -0,0 +1,212 @@
...
+            <img
+              src={project_details.images[0]}
+              alt='bacpack'
+              className='object-cover rounded-lg h-full w-full mx-5 my-5'
+            />
+          </div>
+
+          <div className='mr-5 h-28 mt-5'>
+            <div className='no-scrollbar flex ml-4 h-[112px] no-scrollbar overflow-x-scroll'>
+              {project_details.images.map(item => (
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: You are already showing the first project picture above, don't do it here too.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

**Comment by iamalextodoran** [comment]
GF-13-17-React Dom and Project page --> GF-13, GF-17 - React Route Dom and Project page

If you implement multiple tasks on the same PR, they are sorted by ticket id and separated by commas.

**Review by MIKALAKE — APPROVED** [review]

File: package.json [hunk 16]
```diff
@@ -6,6 +6,7 @@
     "@testing-library/jest-dom": "5.16.4",
     "@testing-library/react": "13.3.0",
     "@testing-library/user-event": "13.5.0",
+    "axios": "^0.27.2",
```
Comment (MIKALAKE) [inline package.json]: `^` We want to update versions

Comment (iamalextodoran) [reply package.json]: It's not even supposed to go on this PR.

File: src/show/pages/public/Project/index.js [hunk 17]
```diff
@@ -0,0 +1,233 @@
...
+};
+
+const Project = () => {
+  return (
+    <div className='flex flex-col bg-white h-full w-full'>
+      <div className='flex bg-white h-20 mt-5'>
+        <div className='flex h-20 w-20 justify-center items-center'>
+          <div className='flex items-center justify-center w-16 h-16 rounded-full ml-5'>
+            <img
+              src={projectDetails.team_members[0].avatar_url}
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Put the first team member into a variable, and do the same with the rest of the team members.

File: src/show/pages/public/Project/index.js [hunk 18]
```diff
@@ -0,0 +1,233 @@
...
+            <img
+              src={projectDetails.images[0]}
+              alt='bacpack'
+              className='object-cover rounded-lg h-full w-full mx-5 my-5'
+            />
+          </div>
+
+          <div className='mr-5 h-28 mt-5'>
+            <div className='no-scrollbar flex ml-4 h-[112px] no-scrollbar overflow-x-scroll'>
+              {projectDetails.images.map(item => (
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Show all the images, but the first. And put them into a variable

File: src/show/pages/public/Project/index.js [hunk 19]
```diff
@@ -0,0 +1,233 @@
...
+        <div className='flex flex-col justify-center ml-2 w-full text-2xl'>
+          {projectDetails.statement}
+        </div>
+      </div>
+
+      <div className='flex mt-5 mb-10'>
+        <div className='flex flex-col h-112 w-3/5'>
+          <div className='flex justify-center items-center h-full w-full'>
+            <img
+              src={projectDetails.images[0]}
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Put this into a variable

File: src/show/pages/public/Project/index.js [hunk 20]
```diff
@@ -0,0 +1,233 @@
...
+          <div className='flex justify-center'>
+            <div className='flex flex-col'>
+              <div className='flex justify-center text-2xl mt-3'>
+                Team members
+              </div>
+              <div className='flex flex-col'>
+                <div className='flex justify-center'>
+                  <div className='flex h-20 w-20 justify-center items-center'>
+                    <div className='flex items-center justify-center w-16 h-16 rounded-full'>
+                      {projectDetails.team_members.slice(2, 3).map()}
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Put this in a variable.

File: src/show/pages/public/Project/index.js [hunk 21]
```diff
@@ -0,0 +1,233 @@
...
+                    {projectDetails.team_members
+                      .slice(2, 3)
+                      .map(member => member.name)}
+                  </div>
+                </div>
+                <div className='flex'>
+                  <div>
+                    <div className='flex h-20 w-20 justify-center items-center'>
+                      <div className='flex items-center justify-center w-16 h-16 rounded-full'>
+                        {projectDetails.team_members.slice(1, 2).map(member => (
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Put this in a variable.

File: src/show/pages/public/Project/index.js [hunk 22]
```diff
@@ -0,0 +1,233 @@
...
+                            alt={member.name}
+                            className='rounded-full object-cover h-14 w-14'
+                          />
+                        ))}
+                      </div>
+                    </div>
+                    <div className='flex justify-center'>
+                      {projectDetails.team_members
+                        .slice(1, 2)
+                        .map(member => member.name)}
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Put this in a variable.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]
Modify the PR title.

And as we spoke, put the first team member, the CEO in one line and a variable, and the rest of the team in the next line (in the UI) and in a different variable.

File: src/show/pages/public/Project/index.js [hunk 23]
```diff
@@ -0,0 +1,208 @@
...
+            <div className='mt-5 px-5 text-lg text-center'>
+              {projectDetails.progress_goal}$ / {projectDetails.goal}$
+            </div>
+            <div className='px-5'>
+              <ProgressBar percentage={25} />
+            </div>
+            <div className='flex items-center justify-center mt-5'>
+              <Button
+                className='flex w-1/6 justify-center'
+                variant='invertoutline'
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Your new variant type is `invert-outline`, use it wisely.

File: src/show/pages/public/Project/index.js [hunk 24]
```diff
@@ -0,0 +1,208 @@
...
+                    </div>
+                  </div>
+                  <div className='flex justify-center items-center'>
+                    {firstMember.name}
+                  </div>
+                </div>
+                <div className='flex flex-col w-96 px-0 mx-0'>
+                  <div className='flex justify-center items-center'>
+                    {restMembers.map((member, index) => (
+                      <div className='flex flex-col items-center'>
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Your are mapping through an array of members yet I cannot see a key prop so React knows what to re-render.

File: src/show/pages/public/Project/index.js [hunk 25]
```diff
@@ -0,0 +1,208 @@
...
+                  </div>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+        <div className='flex w-2/5'>
+          <div className='mx-5 w-full'>
+            <img
+              src={projectDetails.images[1]}
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Why only show the second image in this array?

File: src/show/pages/public/Project/index.js [hunk 26]
```diff
@@ -0,0 +1,208 @@
...
+              alt='bacpack'
```
Comment (iamalextodoran) [inline src/show/pages/public/Project/index.js]: Typo.

**Review by iamalextodoran — CHANGES_REQUESTED** [review]

**Review by iamalextodoran — APPROVED** [review]

**Pull request merged**